include/oneapi/internal/execution_context_sycl.h

```diff
@@ -25,6 +25,7 @@
 #include <CL/sycl.hpp>
```

Comment:
@michael-smirnov, can you please clarify why removing SYCL headers is necessary?

Reply:
@rlnx my question is mostly related to `<vector>` and `<cstring>` includes, because first one is prohibited and the second is unused. What is up to `<CL/sycl.hpp>` header, it is included with the use of "daal_sycl.h" - probably this is the right place to put SYCL/OpenCL related headers?

Reply:
@mchernov-intel @rlnx what a decision we are accepting regarding OpenCL/SYCL includes?